client/client_test.go

```diff
@@ -765,6 +765,24 @@ func TestClient_SaveRestoreState(t *testing.T) {
 						return fmt.Errorf("expected running client status, got %v",
 							ar.AllocState().ClientStatus)
 					}
+
```

Comment:
Note for reviewers: this test was actually already racy but it's a _very_ tight race before the changes in this PR. So these fixes for the test remove the race.